core/lib/utils/list_utils.dart

```diff
@@ -0,0 +1,14 @@
+List<List<T>> partition<T>(List<T> list, bool Function(T) predicate) {
```

Comment:
```suggestion
Tupple2<List, List> partition<T>(List<T> list, bool Function(T) validate) {
```

Reply:
because this function only separated list into 2 list, why not return a tuple?

Reply:
done